WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsCollectionViewFlowLayout.swift

```diff
@@ -0,0 +1,199 @@
+        var previousFrame: CGRect = .zero
+
+        let numberOfItems: Int = collectionView.numberOfItems(inSection: 0)
+
+        for item in 0 ..< numberOfItems {
+            let indexPath: IndexPath = IndexPath(row: item, section: 0)
+
+            let itemSize = sizeForItem(at: indexPath)
+            var frame: CGRect = CGRect(origin: .zero, size: itemSize)
+
+            if item == 0 {
+                let minX: CGFloat = isRightToLeft ? maxContentWidth - frame.width : 0
+                frame.origin = CGPoint(x: minX, y: contentInsets.top)
+            } else {
+                frame.origin.x = isRightToLeft ? previousFrame.minX - itemSpacing - frame.width : previousFrame.maxX + itemSpacing
+
+                // If the new X position will go off screen move it to the next row
+                let needsNewRow = isRightToLeft ? frame.origin.x < 0 : frame.maxX > maxContentWidth
+                if needsNewRow {
+                    frame.origin.x = isRightToLeft ? (maxContentWidth - frame.width) : 0
+                    currentRow += 1
+                }
+                frame.origin.y = currentRow * (cellHeight + itemSpacing) + contentInsets.top
+            }
+
+            let attribute = UICollectionViewLayoutAttributes(forCellWith: indexPath)
+            attribute.frame = frame
+
+            layoutAttributes.append(attribute)
+
+            previousFrame = frame
+        }
+
+        // Update content size
+        contentSize.width = maxContentWidth
+        contentSize.height = currentRow * (cellHeight + itemSpacing) + contentInsets.top + contentInsets.bottom
+    }
+
+    override func invalidateLayout() {
+        contentSize = .zero
+        layoutAttributes = []
+
+        super.invalidateLayout()
+    }
+
+    /// Get the size for the given index path from the delegate, or the default item size
+    /// - Parameter indexPath: index path for the item
+    /// - Returns: The width for the cell either from the delegate or the itemSize property
+    private func sizeForItem(at indexPath: IndexPath) -> CGSize {
+        guard
+            let collectionView = collectionView,
+            let delegate = collectionView.delegate as? UICollectionViewDelegateFlowLayout,
+            delegate.responds(to: #selector(delegate.collectionView(_:layout:sizeForItemAt:)))
+        else {
+            return CGSize(width: itemSize.width, height: cellHeight)
+        }
+
+        let size = delegate.collectionView!(collectionView, layout: self, sizeForItemAt: indexPath)
+        return CGSize(width: size.width, height: cellHeight)
+    }
+
+    override open func layoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        if collectionView?.traitCollection.horizontalSizeClass == .regular {
+            return centeredLayoutAttributesForElements(in: rect)
+        }
+
+        var layoutAttributes = [UICollectionViewLayoutAttributes]()
+
+        for attribute in self.layoutAttributes {
```

Comment:
Swifting ;) 
This whole for could be:
 let layoutAttributes: [UICollectionViewLayoutAttributes] = self.layoutAttributes.filter {
             return $0.frame.intersects(rect)
        }

Reply:
Good plan, with that I don't need to define the layoutAttributes variable anymore so I'll just return the filter directly.